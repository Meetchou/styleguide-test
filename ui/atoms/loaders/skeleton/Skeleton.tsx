import React, { FC } from 'react';
import cls from 'classnames';

import { SizeType } from '@ui/types';
import { sizes as buttonSizes } from '@ui/atoms/buttons/button/utils';

export type { SizeType };

export const sizes = {
  small: 'h-3',
  base: 'h-6',
  large: 'h-10'
};

export type SkeletonProps = {
  className?: string;
  size?: SizeType;
  button?: boolean;
  field?: boolean;
};

const Skeleton:FC<SkeletonProps> = ({
  className,
  size,
  button,
  field
}) => {
  if (!size && (button || field)) {
    // in button or field case we force default size
    size = 'base';
  }
  const sizesValues = button ? buttonSizes : sizes;

  return (
    <div className={cls(
      'animate-pulse bg-gray-100',
      {
        'rounded-lg': button || field,
        'rounded': !button && !field
      },
      size ? sizesValues[size] : '',
      className
    )} />
  );
};

export default Skeleton;
