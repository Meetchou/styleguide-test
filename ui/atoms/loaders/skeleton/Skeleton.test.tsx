import React from 'react';

import render from '@ui/testing/render';

import Skeleton from './Skeleton';

describe('ui/atoms/loaders/skeleton/Skeleton', () => {
  describe('renders correctly', () => {
    it.each([
      {},
      { button: true },
      { field: true }
    ])('%p', opts => {
      const tree = render(<Skeleton {...opts} />);

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
