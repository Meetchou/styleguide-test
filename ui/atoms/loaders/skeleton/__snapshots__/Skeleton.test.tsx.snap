// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/loaders/skeleton/Skeleton renders correctly {"button": true} 1`] = `
<div
  className="animate-pulse bg-gray-100 rounded-lg h-9"
/>
`;

exports[`ui/atoms/loaders/skeleton/Skeleton renders correctly {"field": true} 1`] = `
<div
  className="animate-pulse bg-gray-100 rounded-lg h-6"
/>
`;

exports[`ui/atoms/loaders/skeleton/Skeleton renders correctly {} 1`] = `
<div
  className="animate-pulse bg-gray-100 rounded"
/>
`;
